src/cns.ts

```diff
@@ -54,7 +56,7 @@ export default class Cns extends EthereumNamingService {
    * @returns- Returns a promise that resolves in an object
    */
   async resolve(domain: string): Promise<ResolutionResponse> {
-    throw new Error('This method is unsupported for CNS');
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {method: this.name});
```

Comment:
In any programming language a change like this one is not considered breaking because people are checking error with `instanceof` in the `catch` block. Usually using built-in language tools. I personally made a ton of those in ruby and java, and no one ever complained. I think we can accept this as a non-breaking change in JS even though exception handling is terrible in JS world.

Reply:
Yes, but in this case the only way to catch an error was to match its message against a string, and we're changing it